test/unit/toggle_task.ts

```diff
@@ -0,0 +1,132 @@
+import { expect } from "chai";
+import * as todoer from "../../src/core/todoer";
+
+describe("Toggling a Task", () => {
+  describe("Should toggle non-tasks to task", () => {
+    // prettier-ignore
+    const theories = [
+      [ "Go shopping",       "- [ ] Go shopping"    ],
+      [ " \tGo shopping\t ", "- [ ] Go shopping"    ],
+      [ "[] Go shopping",    "- [ ] [] Go shopping" ],
+      [ "",                  "- [ ] "               ],
+      [ "\t",                "- [ ] "               ],
+      [ " ",                 "- [ ] "               ],
+    ];
+
+    for (const [input, expected] of theories) {
+      it(`Should toggle ${JSON.stringify(input)}`, () => {
+        const actual = todoer.toggleTasks(input);
+        expect(actual).to.be.equal(expected);
+      });
+    }
+  });
+
+  describe("Should toggle incomplete tasks to non-tasks", () => {
+    // prettier-ignore
+    const theories = [
+      [ "- [ ] Go shopping",       "Go shopping"     ],
+      [ "- [ ] ~Go shopping~",     "~Go shopping~"   ],
+      [ "- [ ] Go ~~shopping~~",   "Go ~~shopping~~" ],
+      [ "- [ ] ~~Go shopping~~",   "~~Go shopping~~" ],
+      [ "- [ ] [ ] Go shopping",   "[ ] Go shopping" ],
+      [ "- [ ] \t Go shopping \t", "Go shopping"     ],
+      [ "- [ ] ",                  ""                ],
+    ];
+
+    for (const [input, expected] of theories) {
+      it(`Should toggle ${JSON.stringify(input)}`, () => {
+        const actual = todoer.toggleTasks(input);
+        expect(actual).to.be.equal(expected);
+      });
+    }
+  });
+
+  describe("Should toggle completed tasks to non-tasks", () => {
+    // prettier-ignore
+    const theories = [
```

Comment:
Does this functinoality make sense to you?  If you have a completed task, you can turn it back into a non-task?

Reply:
Maybe have to uncomplete it first? Once we add things like startTime and endTime, this could potentially wipe out stats.